### ISSUE TITLE
HMAccessorySetupManager removed from Xamarin.iOS 16

### ISSUE THREAD

**Issue by johannes-keinestam** [xamarin/xamarin-macios#16415]
Since the `HMHome.AddAndSetupAccessoriesAsync` method has been [deprecated](https://developer.apple.com/documentation/homekit/hmhome/2941039-addandsetupaccessories) as of iOS 15.4, `HMAccessorySetupManager.PerformAccessorySetupAsync` must be used instead. This worked fine in Xamarin.iOS 15.12.0.2 which I'm running, but a member of my team has updated to Xamarin.iOS 16.0.0.72 which seems to throw an exception saying that the API in question has been removed, but I cannot find any information in Apples documentation to support that.

### Steps to Reproduce

1. Add code using the `HMAccessorySetupManager` class, e.g.:
```
var setupRequest = new HMAccessorySetupRequest { Payload = string.Empty };

var hm = new HMAccessorySetupManager();
await hm.PerformAccessorySetupAsync(setupRequest);
```

2. Run the app on an iOS device.

### Expected Behavior
`HMAccessorySetupManager` is available to use (see [Apple docs](https://developer.apple.com/documentation/homekit/hmaccessorysetupmanager), where the API is not deprecated)

### Actual Behavior
Using the `HMAccessorySetupManager` constructor throws an exception:
```
System.InvalidOperationException: This API has been removed from the 'HomeKit' framework.
  at HomeKit.HMAccessorySetupManager..ctor () [0x0000b] in /Library/Frameworks/Xamarin.iOS.framework/Versions/16.0.0.72/src/Xamarin.iOS/HomeKit/HMCompat.cs:116 
  at MyApp.iOS.HomeKitSetupController.Setup (System.String setupPayload) [0x000be] in /Users/user/my-app/iOS/HomeKitSetupController.cs:177
```

### Environment

<details>
<summary>Version information</summary>

```
Visual Studio Professional 2022 for Mac Preview
Version 17.4 Preview (17.4 build 2326)
Installation UUID: b5eac099-3f79-40bf-a1d8-5d720796ce85

Runtime
.NET 6.0.9 (64-bit)
Architecture: Arm64

Roslyn (Language Service)
4.4.0-3.22505.14+53c4590aabe91a32e33324c5095f7e9ee41dbf78

NuGet
Version: 6.3.1.1

.NET SDK (Arm64)
SDK: /usr/local/share/dotnet/sdk/7.0.100-rc.2.22477.23/Sdks
SDK Versions:
	7.0.100-rc.2.22477.23
	6.0.402
	6.0.401
	6.0.400
	6.0.400-preview.22330.6
	6.0.301
	6.0.300
	6.0.203
	6.0.202
	6.0.201
	6.0.101
	6.0.100

MSBuild SDKs: /Applications/Visual Studio (Preview).app/Contents/MonoBundle/MSBuild/Current/bin/Sdks
.NET SDK (x64)
SDK Versions:
	6.0.402
	6.0.401
	6.0.400
	6.0.400-preview.22330.6
	6.0.301
	6.0.300
	6.0.203
	6.0.202
	6.0.201
	6.0.106
	6.0.103
	6.0.102
	6.0.101
	5.0.408
	5.0.407
	5.0.406
	5.0.405
	5.0.404
	4.1.424
	3.1.423
	3.1.422
	3.1.421
	3.1.420
	3.1.419
	3.1.418
	3.1.417
	3.1.416

.NET Runtime (Arm64)
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	7.0.0-rc.2.22472.3
	6.0.10
	6.0.9
	6.0.8
	6.0.6
	6.0.5
	6.0.4
	6.0.3
	6.0.1
	6.0.0

.NET Runtime (x64)
Runtime: /usr/local/share/dotnet/x64/dotnet
Runtime Versions:
	6.0.10
	6.0.9
	6.0.8
	6.0.6
	6.0.5
	6.0.4
	6.0.3
	6.0.2
	6.0.1
	5.0.17
	5.0.16
	5.0.15
	5.0.14
	5.0.13
	3.1.30
	3.1.29
	3.1.28
	3.1.27
	3.1.26
	3.1.25
	3.1.24
	3.1.23
	3.1.22

Xamarin.Profiler
Version: 1.8.0.19
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Apple Developer Tools
Xcode 14.0.1 (21336)
Build 14A400

Xamarin.Mac
Version: 8.12.0.2 (Visual Studio Professional)
Hash: 87f98a75e
Branch: d17-3
Build date: 2022-07-25 20:18:54-0400

Xamarin.iOS
Version: 16.0.0.72 (Visual Studio Professional)
Hash: 6756a1146
Branch: release/6.0.4xx-xcode14
Build date: 2022-09-21 08:51:06-0400

Xamarin Designer
Version: 17.4.0.133
Hash: 8da0a9c709
Branch: remotes/origin/d17-4
Build date: 2022-10-13 22:38:00 UTC

Xamarin.Android
Version: 13.1.0.1 (Visual Studio Professional)
Commit: xamarin-android/d17-4/13ba222

Android SDK: /Users/user/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		12.0 (API level 31)
		11.0 (API level 30)
		9.0  (API level 28)
		13.0 (API level 33)

SDK Command-line Tools Version: 7.0
SDK Platform Tools Version: 33.0.2
SDK Build Tools Version: 33.0.0

Build Information:
Mono: a96bde9
Java.Interop: xamarin/java.interop/d17-4@fcc33ce2
SQLite: xamarin/sqlite/3.39.3@23e1ae7
Xamarin.Android Tools: xamarin/xamarin-android-tools/main@0be567a

Microsoft Build of OpenJDK
Java SDK: /Library/Java/JavaVirtualMachines/microsoft-11.jdk
11.0.12
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Eclipse Temurin JDK
Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk
1.8.0.302
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 17.4.0.54
Hash: 6eabb9e
Branch: remotes/origin/d17-4
Build date: 2022-10-13 22:38:04 UTC

Android Device Manager
Version: 0.0.0.1206
Hash: 886af39
Branch: 886af39
Build date: 2022-10-13 22:38:04 UTC

Build Information
Release ID: 1704002326
Git revision: 5cebb8f0d1a2529ee4549db8bf35550599378c80
Build date: 2022-10-13 22:35:45+00
Build branch: release-17.4
Build lane: release-17.4

Operating System
Mac OS X 12.6.0
Darwin 21.6.0 Darwin Kernel Version 21.6.0
```

</details>

**Comment by dalexsoto** [comment]
You seem to be right 

https://developer.apple.com/documentation/homekit/hmaccessorysetupmanager/3920432-performaccessorysetupusingreques?language=objc

But also our submission tests told us that Apple was rejecting that selector (hence the deprecation) https://github.com/xamarin/xamarin-macios/commit/8a07db61bbfaff7e09bd2a055fcfd515473a96bb it may have been a hiccup, we'll have a look

**Comment by dalexsoto** [comment]
This will be fixed when we publish Xcode 14.1 compatible bits, Thanks for the report!
